backend/internal/auth/sessionmanager.go

```diff
@@ -0,0 +1,46 @@
+package auth
+
+import (
+	"sync"
+	"time"
+
+	"github.com/ibeckermayer/teleport-interview/backend/internal/model"
+)
+
+// Session is an individual user's session
+type Session struct {
+	SessionID SessionID
+	Account   model.Account
+	Expires   time.Time
+}
+
+// SessionManager is an in-memory session store. The app should only ever create one
+// of these and pass it around as a pointer
+type SessionManager struct {
+	store   map[SessionID]Session
+	timeout time.Duration // absolute timeout for individual sessions
+	mtx     sync.RWMutex
```

Comment:
```suggestion
	mtx     sync.Mutex
```

Reply:
My thinking is that I'd get some performance improvement if I use `RLock()` for reading a session from `sm.store["sessionid"]` (into, say, an auth protected `http.Handler`) and then `Lock()` if/when i need to update the session.

Reply:
Ah, right, you'll be adding the reading part in a future PR.
RWMutex makes sense.